client/src/app/containers/Game/useGame.ts

```diff
@@ -0,0 +1,235 @@
+import axios from 'axios';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { useAuth } from '../../../auth/useAuth';
+import { RootState } from '../../../store/store';
+import { useAlert } from '../Alert/useAlert';
+import { Card, Game, Phase } from './models';
+import {
+    actions,
+    CountCribAction,
+    CountHandAction,
+    CribAction,
+    CutAction,
+    DealAction,
+    GameAction,
+    PegAction,
+} from './slice';
+import { CreateGameResponse } from './types';
+
+interface Result {
+    game: Game;
+    selectedCards: Card[];
+    createGame: (playerIDs: string[]) => Promise<void>;
+    getGame: (id: number) => Promise<void>;
+    refreshGame: () => Promise<void>;
+    toggleSelectedCard: (c: Card) => void;
+    clearSelectedCards: () => void;
+    submitDealAction: (a: DealAction) => Promise<void>;
+    submitBuildCribAction: (a: CribAction) => Promise<void>;
+    submitCutDeckAction: (a: CutAction) => Promise<void>;
+    submitPegAction: (a: PegAction) => Promise<void>;
+    submitCountHandAction: (a: CountHandAction) => Promise<void>;
+    submitCountCribAction: (a: CountCribAction) => Promise<void>;
+}
+// getPlayerAction returns the JSON struct which the server knows
+// how to interpret
+interface ServerCard {
+    s: number;
+    v: number;
+}
+interface ActionRequest {
+    pID: string;
+    gID: number;
+    o: ReturnType<typeof mapPhaseToOverComes>;
+    a: {
+        ns?: number;
+        cs?: ServerCard[];
+        p?: number;
+        sg?: boolean;
+        c?: ServerCard;
+        pts?: number;
+    };
+}
+
+const mapPhaseToOverComes = (p: Phase) => {
+    if (p === 'Deal') {
+        return 0;
+    }
+    if (p === 'BuildCrib') {
+        return 1;
+    }
+    if (p === 'Cut') {
+        return 2;
+    }
+    if (p === 'Pegging') {
+        return 3;
+    }
+    if (p === 'Counting') {
+        return 4;
+    }
+    if (p === 'CribCounting') {
+        return 5;
+    }
+    return -1;
+};
+
+const cardToGolangCard = (c: Card): ServerCard => {
```

Comment:
:book: `convertToServerCard`?

Reply:
Hey man, you named it ;) Yeah, I'll change it